lib/agenda.js

```diff
@@ -857,17 +949,22 @@ function processJobs(extraJob) {
 
     // Don't lock because of a limit we have set (lockLimit, etc)
     if (!shouldLock(name)) {
+      debug('lock limit reached in queue filling for [%s]', name);
       return;
     }
 
     // Set the date of the next time we are going to run _processEvery function
     var now = new Date();
     self._nextScanAt = new Date(now.valueOf() + self._processEvery);
 
+    // NOTE: This may print a lot based on _processInterval
+    debug('_nextScanAt set to [%s]', self._nextScanAt.toISOString());
```

Comment:
Let's leave this in for now and see how it does. I'm guessing this would only really be a problem with short process intervals?